src/TapeChart.js

```diff
@@ -5,47 +5,55 @@ class TapeChart {
     this.bookings = bookings;
   }
 
-getAvailableRoomsByDate(date) {
+  getAvailableRooms(date, roomType) {
     let availableByDate = this.rooms.reduce((acc, room) => {
       let bookingsByRoom = this.bookings.filter(booking =>
         Number(booking.roomNumber) === room.number);
       if (!bookingsByRoom.some(booking => booking.date === date)) {
         acc.push(room);
       }
-        return acc;
-    }, [])
-  return availableByDate;
-}
+      return acc;
+    }, []);
+    if (roomType && roomType !== 'any') {
+      return this.filterByRoomType(availableByDate, roomType);
+    } else {
+      return availableByDate;
+    }
+  }
 
-getOccupancy(date) {
-  return 100 - (this.getAvailableRoomsByDate(date).length
-    / this.rooms.length * 100);
-}
+  filterByRoomType(roomList, roomType) {
+    return roomList.filter(room => room.roomType === roomType)
+  }
 
-filterBookingsByMetric(metric, spec) {
-  return this.bookings.filter(booking => booking[metric] === spec);
-}
+  getOccupancy(date) {
+    return 100 - (this.getAvailableRooms(date).length
+      / this.rooms.length * 100);
+  }
 
-getDaysRevenue(metric, spec) {
-  let bookingsByDate = this.filterBookingsByMetric(metric, spec);
-  return Number(bookingsByDate.reduce((acc, booking) => {
-    this.rooms.forEach(room => {
-      if (room.number === Number(booking.roomNumber)) {
-        acc += room.costPerNight;
-      }
-    })
-    return acc;
-  }, 0).toFixed(2));
-}
+  filterBookingsByMetric(metric, spec) {
+    return this.bookings.filter(booking => booking[metric] === spec);
+  }
 
-filterAvailableRoomsByFeature(date, feature, spec) {
-  let availableRoomsByDate = this.getAvailableRoomsByDate(date);
-  return availableRoomsByDate.filter(room => room[feature] === spec);
-}
+  getDaysRevenue(metric, spec) {
```

Comment:
When you call this the metric and spec are going to be dates?

Reply:
Thanks for pointing this out. 

`metric` would be pass either 'date' or 'userID' and `spec` would be passed the specific date or userID.

 I built this dynamically so that it could potentially inherited and reused by my User class.  But I built a separate very similar one in User.

I got caught up on deciding my classes and inheritance and just started building.  I'll open an issue to make sure I refactor these correctly once I have my head wrapped around the inheritance piece.